test/libsolidity/NatspecJSONTest.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/**
+ * Unit tests for the solidity compiler ABI JSON Interface output.
+ */
+
+#include <test/libsolidity/NatspecJSONTest.h>
+
+#include <libsolutil/CommonIO.h>
+#include <libsolutil/StringUtils.h>
+
+#include <boost/algorithm/string/predicate.hpp>
+
+#include <fmt/format.h>
+
+#include <vector>
+
+using namespace std;
+using namespace solidity::frontend::test;
+using namespace solidity::util;
+
+ostream& solidity::frontend::test::operator<<(ostream& _output, NatspecJSONKind _kind)
+{
+	switch (_kind) {
+	case NatspecJSONKind::Devdoc: _output << "devdoc"; break;
+	case NatspecJSONKind::Userdoc: _output << "userdoc"; break;
```

Comment:
Yes. I actually don't like that style too much (I'd prefer the `cases` to be indented one level), but I stick to it because that's the style used around the code base.